Sources/Prometheus/PrometheusCollectorRegistry.swift

```diff
@@ -409,6 +409,10 @@ public final class PrometheusCollectorRegistry: Sendable {
                 let labelsKey = LabelsKey(counter.labels)
                 guard dimensions[labelsKey] === counter else { return }
                 dimensions.removeValue(forKey: labelsKey)
+                guard !dimensions.isEmpty else {
+                    store.removeValue(forKey: counter.name)
+                    return
+                }
                 store[counter.name] = .counterWithLabels(labelNames, dimensions)
```

Comment:
Personal preference nit, can we write his as:
```suggestion
                if dimensions.isEmpty {
                    store.removeValue(forKey: counter.name)
                } else {
                    store[counter.name] = .counterWithLabels(labelNames, dimensions)
                }
```
This way we remove a double negation, which should make this easier to reason about.

Reply:
I followed the double-negation guard pattern from the rest of this file (used in all the make APIs), e.g. 

```swift 
        guard !labels.isEmpty else {
            return self.makeValueHistogram(name: name, buckets: buckets)
        }
```

But I don't love it, so I'm happy to switch to an `if-else`.